parsl/providers/base.py

```diff
@@ -1,124 +1,13 @@
-import os
 from abc import ABCMeta, abstractmethod, abstractproperty
-from enum import IntEnum
 import logging
 from typing import Any, Dict, List, Optional
 
 from parsl.channels.base import Channel
+from parsl.jobs.states import JobStatus
```

Comment:
Similar suggestion as for `executor.base`: this doesn't appear to be utilized in this file, so can discourage imports of this class from this module by wrapping it in the type checking conditional:
```python
if t.TYPE_CHECKING:
    from parsl.jobs.states import JobStatus
```